src/kubernetes/reflector.rs

```diff
@@ -30,14 +33,17 @@ pub async fn custom_reflector<K, W>(
                     Some(Ok(event)) => {
                         match event {
                             // Immediately reoncile `Applied` event
-                            watcher::Event::Applied(_) => {
+                            watcher::Event::Applied(ref obj) => {
                                 trace!(message = "Processing Applied event.", ?event);
                                 store.apply_watcher_event(&event);
+                                let meta_descr = MetaDescribe::from_meta(obj.meta());
+                                meta_cache.store(meta_descr, true);
                             }
                             // Delay reconciling any `Deleted` events
-                            watcher::Event::Deleted(_) => {
-                                trace!(message = "Queuing Deleted event.", ?event);
+                            watcher::Event::Deleted(ref obj) => {
                                 delay_queue.insert(event.to_owned(), delay_deletion);
+                                let meta_descr = MetaDescribe::from_meta(obj.meta());
+                                meta_cache.store(meta_descr, false);
                             }
                             // Clear all delayed events on `Restarted` events
                             watcher::Event::Restarted(_) => {
```

Comment:
Consider clearing the meta cache when receiving Restarted

Reply:
Added it